### PR TITLE
[OPPRO-206] Keep scan+filter pattern for arrow backend

### PR DESCRIPTION
**Pull request by marin-ma**

### CODE REVIEW

File: jvm/src/main/scala/io/glutenproject/expression/ConverterUtils.scala [hunk 1]
```diff
@@ -261,6 +261,8 @@ object ConverterUtils extends Logging {
           typedFuncName.concat("date")
         case StringType =>
           typedFuncName.concat("str")
+        case DecimalType() =>
+          typedFuncName.concat("decimal")
```
Comment (rui-mo) [inline jvm/src/main/scala/io/glutenproject/expression/ConverterUtils.scala:265]: Maybe change to "dec" for decimal to be consistent with Substrait upstream.
Reference: https://substrait.io/extensions/#function-signature-compound-names

File: cpp/CMakeLists.txt [hunk 2]
```diff
@@ -46,7 +46,7 @@ set(root_directory ${PROJECT_BINARY_DIR})
 #
 
 if(DEBUG)
-  add_definitions(-DDEBUG)
+  add_definitions(-DGLUTEN_PRINT_DEBUG)
```
Comment (jinchengchenghh) [inline cpp/CMakeLists.txt:49]: Do you think we need to set two debug option?

Comment (marin-ma) [reply cpp/CMakeLists.txt:49]: No. DEBUG is cmake flag and GLUTEN_PRINT_DEBUG is the compilation flag. I just found the DEBUG compilation flag also used in velox. Change it to another one can avoid conflict.

**Comment by marin-ma** [comment]
@rui-mo @jinchengchenghh Do you have any further comments?

**Comment by rui-mo** [comment]
LGTM.

**Comment by jinchengchenghh** [comment]
LGTM

**Review by zzcclp — APPROVED** [review]
LGTM, will be merged.

**Pull request merged**